kore/src/Kore/Builtin/List.hs

```diff
@@ -446,23 +442,20 @@ unifyEquals
                                 builtin1
                                 x2
                                 builtin2
+                                & lift
                     _ -> empty
-        unifyEquals0 dv1@(InternalList_ builtin1) pat2 =
+        unifyEquals0 (InternalList_ builtin1) pat2 =
             case pat2 of
                 InternalList_ builtin2 ->
                     lift $ unifyEqualsConcrete builtin1 builtin2
-                app@(App_ symbol2 args2)
+                (App_ symbol2 args2)
                     | isSymbolConcat symbol2 ->
-                        lift $ case args2 of
+                        case args2 of
                             [InternalList_ builtin2, x@(Var_ _)] ->
-                                unifyEqualsFramedRight builtin1 builtin2 x
+                                unifyEqualsFramedRight builtin1 builtin2 x & lift
                             [x@(Var_ _), InternalList_ builtin2] ->
-                                unifyEqualsFramedLeft builtin1 x builtin2
-                            [_, _] ->
-                                Builtin.unifyEqualsUnsolved
-                                    simplificationType
-                                    dv1
-                                    app
+                                unifyEqualsFramedLeft builtin1 x builtin2 & lift
+                            [_, _] -> empty
```

Comment:
How did the old case return `\\bottom`?

Reply:
The old version called `lift empty` instead of `empty`.